cli/tests/pcluster/createami/test_pcluster_createami.py

```diff
@@ -116,3 +116,41 @@ def test_validate_createami_args_architecture_compatibility(
 
     if instance_type is None or base_ami_architecture in supported_instance_archs:
         createami.utils.get_supported_os_for_architecture.assert_called_with(base_ami_architecture)
+
+
+@pytest.mark.parametrize(
+    "post_install_script_url, tmp_dir",
+    [
+        ("https://bucket.s3.us-east-1.amazonaws.com/script.sh", "/tmp"),
+        ("http://script.sh", "/tmp"),
+        ("file://script.sh", "/tmp"),
+        # test for error
+        ("script", "/tmp"),
+        # test for script was not specified
+        (None, "/tmp"),
+    ],
+)
+def test_get_post_install_script_dir(mocker, post_install_script_url, tmp_dir):
+    mocker.patch("pcluster.createami.os.mkdir")
+    mocker.patch("pcluster.createami.urlretrieve")
+    mocker.patch("pcluster.createami.copyfile")
+    mocker.patch("pcluster.createami._get_current_timestamp").return_value = "now"
+    if post_install_script_url:
+        expected_script = any(
+            [
+                post_install_script_url.startswith("https://"),
+                post_install_script_url.startswith("http://"),
+                post_install_script_url.startswith("s3://"),
+                post_install_script_url.startswith("file://"),
+            ]
+        )
+        if not expected_script:
+            with pytest.raises(SystemExit) as sysexit:
+                createami._get_post_install_script_dir(post_install_script_url, tmp_dir)
+            assert_that(sysexit.value.code).is_not_equal_to(0)
+        else:
+            assert_that(createami._get_post_install_script_dir(post_install_script_url, tmp_dir)).is_equal_to(
+                os.path.join("/tmp/script", "now-script.sh")
+            )
+    else:
+        assert_that(createami._get_post_install_script_dir(post_install_script_url, tmp_dir)).is_equal_to(None)
```

Comment:
after addressing the comments, the test can be probably simplified in something like
```
if is_valid_url
   assert_that(createami._get_post_install_script_dir(post_install_script_url, tmp_dir)).is_equal_to( 
                os.path.join("/tmp/script", "now-script.sh") if post_install_script_url else None
else
    assert_that error is raised
```

Reply:
Done